### DISCUSSION TITLE
"Enforce required keys and enforce defaults" makes code needlessly cumbersome

### DISCUSSION THREAD

**Discussion by curieux** [developmentseed/geojson-pydantic#124] [General]
Regarding the change issued here 
https://github.com/developmentseed/geojson-pydantic/blob/main/CHANGELOG.md#060a0---2023-04-04
```
# Before
Feature(geometry=Point(coordinates=(0,0)))

# Now
Feature(
    type="Feature",
    geometry=Point(
        type="Point",
        coordinates=(0,0)
    ),
    properties=None,
)
```

This change is unfortunate for a few reasons:
- it violates semantic versioning, because the fundamental call has changed in an incompatible way without a major version change https://semver.org/  
- it is incorrect to say that the old system did not follow the GeoJSON spec.  The GeoJSON spec is about JSON, not about Python objects.  As long as the `@validator` and `.json()` work correctly, GeoJSON is met.
- it is unpythonic to repeat identical information `Feature(type='Feature", ...`

We have been using geojson-pydantic for several projects, and have had to issue patches (or pin to <0.6.0) throughout.

**Comment by vincentsarago** [comment]
Hi @curieux 

Sorry about the frustration the new release is give you. We have been discussing all the point above in the PR and Issue but I'll answer here again.

> it violates semantic versioning, because the fundamental call has changed in an incompatible way without a major version change https://semver.org/

ref: https://github.com/developmentseed/geojson-pydantic/discussions/112

We choose to do a 0.6 because since the beginning of the project we always introduced breaking change in minor version update. I don't think we advertise anywhere that we respect semver 🤷 


 > it is incorrect to say that the old system did not follow the GeoJSON spec. The GeoJSON spec is about JSON, not about Python objects. As long as the @validator and .json() work correctly, GeoJSON is met.

ref: https://github.com/developmentseed/geojson-pydantic/pull/94

We switch the goal of geojson-pydantic a bit to focus on the `validation` part, meaning that you should provide a valid GeoJSON to the Feature/FeatureCollection/Geometries. We understand that this is an important change.

> it is unpythonic to repeat identical information Feature(type='Feature", ...

I'm not sure about pythonic/unpythonic but at least it's better in term of `validation` purpose. If you are looking for a way to construct GeoJSON feature you can maybe use https://github.com/jazzband/geojson

**Reply by curieux** [reply]
Thanks for the response and references.
Regarding semver,  it is in the changelog:
https://github.com/developmentseed/geojson-pydantic/blob/main/CHANGELOG.md#change-log

**Reply by vincentsarago** [reply]
🤦 will remove 😓
